src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/evolutionaryAlgorithm/EvolutionaryAlgorithmQueryOptimizer.java

```diff
@@ -15,90 +15,96 @@
 
 public class EvolutionaryAlgorithmQueryOptimizer implements QueryOptimizer {
     protected final QueryOptimizationContext ctxt;
-    protected final CostModel costModel;
-    private final int nmCandidates;
-    private final int nmSurvivors;
-    private final int nmSteps;
-    protected final RuleApplicationsOfPlan ruleApplicationCache = new RuleApplicationsOfPlan( new RuleInstances() );
+    protected final int nmCandidates;
+    protected final int nmSurvivors;
+    protected final int nmSteps;
 
-    public EvolutionaryAlgorithmQueryOptimizer( final QueryOptimizationContext ctxt, CostModel costModel,
+    protected final RandomizedSelection randomizedSelection = new RandomizedSelection<>();
```

Comment:
Can you please create two separate versions of RandomizedSelection here: one for PhysicalPlanWithCost objects and the other for RuleApplication objects. That way, you can avoid having to cast all the time when using this RandomizedSelection below.